framework/src/modules/dpos_v2/schemas.ts

```diff
@@ -535,3 +535,123 @@ export const genesisStoreSchema = {
 		},
 	},
 };
+
+const delegateJSONSchema = {
+	type: 'object',
+	required: [
+		'name',
+		'totalVotesReceived',
+		'selfVotes',
+		'lastGeneratedHeight',
+		'isBanned',
+		'pomHeights',
+		'consecutiveMissedBlocks',
+	],
+	properties: {
+		name: {
+			type: 'string',
+		},
+		totalVotesReceived: {
+			type: 'string',
+			format: 'uint64',
+		},
+		selfVotes: {
+			type: 'string',
+			format: 'uint64',
+		},
+		lastGeneratedHeight: {
+			type: 'integer',
+			format: 'uint32',
+		},
+		isBanned: {
+			type: 'boolean',
+		},
+		pomHeights: {
+			type: 'array',
+			items: { type: 'integer', format: 'uint32' },
```

Comment:
type: "integer" is probably aligned with JSON Schema standard but is it aligned with our standards? Because as I remember in our codec LIP we stated that type can be either "object" or "array", for other things we need to use "dataType". Or is it different for array items?

Reply:
Endpoint request/response is not encoded using our codec. therefore, the schema is for JSON format

Reply:
I know, but confusing thing is both schemas are living in the same domain, and I am not sure if this is a good decision. Because by using JSON Schema standard over our standard in the same domain(resource path), you break the constraint of codec standard. This has one implication: in future if you want to parse or process all schemas, you set will be the JSON Schema superset for the logic of processing. There is nothing to distinguish a codec schema from endpoint schema here, and it will be a problem if we want distinguish it in the future. I don't know we will ever need it, but I am informing you.

Reply:
we are actually using both already (for example, for the config or other validation)
It's nice to have one schema definition, but currently we need both.
If you want, maybe you can create new issue?

Reply:
I think it's something I should meditate on more, I'll act if there is a need for further action. Resolving for now.